src/composition-engine.js

```diff
@@ -69,6 +69,37 @@ function tryActivateViewModel(context) {
   return context.viewModel.activate(context.model) || Promise.resolve();
 }
 
+class SwapStrategies {
```

Comment:
This is copy-pasted from `templating-router`. We should probably do something about it, like import from `templating` in `templating-router`?